secure-api-gateway-ob-uk-rs-server/src/main/java/com/forgerock/sapi/gateway/ob/uk/rs/server/api/obie/payment/v4_0_0/domesticscheduledpayments/DomesticScheduledPaymentConsentsApiController.java

```diff
@@ -0,0 +1,116 @@
+            OBValidationService<OBWriteDomesticScheduledConsent4> consentValidator,
+            OBWriteDomesticScheduledConsentResponse5Factory consentResponseFactory) {
+
+        this.consentStoreApiClient = consentStoreApiClient;
+        this.consentValidator = consentValidator;
+        this.consentResponseFactory = consentResponseFactory;
+    }
+
+    @Override
+    public ResponseEntity<OBWriteDomesticScheduledConsentResponse5> createDomesticScheduledPaymentConsents(
+            String authorization,
+            String xIdempotencyKey,
+            String xJwsSignature,
+            OBWriteDomesticScheduledConsent4 obWriteDomesticScheduledConsent4,
+            String xFapiAuthDate,
+            String xFapiCustomerIpAddress,
+            String xFapiInteractionId,
+            String xCustomerUserAgent,
+            String apiClientId,
+            HttpServletRequest request,
+            Principal principal) throws OBErrorResponseException {
+
+        logger.info("Processing createDomesticScheduledPaymentConsents request - consent: {}, idempotencyKey: {}, apiClient: {}, x-fapi-interaction-id: {}",
+                obWriteDomesticScheduledConsent4, xIdempotencyKey, apiClientId, xFapiInteractionId);
+
+        consentValidator.validate(obWriteDomesticScheduledConsent4);
+
+        final CreateDomesticScheduledPaymentConsentRequest createRequest = new CreateDomesticScheduledPaymentConsentRequest();
+        createRequest.setConsentRequest(FRWriteDomesticScheduledConsentConverter.toFRWriteDomesticScheduledConsent(obWriteDomesticScheduledConsent4));
+        createRequest.setApiClientId(apiClientId);
+        createRequest.setIdempotencyKey(xIdempotencyKey);
+        createRequest.setCharges(calculateCharges(obWriteDomesticScheduledConsent4));
+
+        final DomesticScheduledPaymentConsent consent = consentStoreApiClient.createConsent(createRequest);
+        logger.info("Created consent - id: {}", consent.getId());
+
+        return new ResponseEntity<>(consentResponseFactory.buildConsentResponse(consent, getClass()), HttpStatus.CREATED);
+    }
+
+    private List<FRCharge> calculateCharges(OBWriteDomesticScheduledConsent4 obWriteDomesticScheduledConsent4) {
+        return Collections.emptyList();
+    }
+
+    @Override
+    public ResponseEntity<OBWriteDomesticScheduledConsentResponse5> getDomesticScheduledPaymentConsentsConsentId(
+            String consentId,
+            String authorization,
+            String xFapiAuthDate,
+            String xFapiCustomerIpAddress,
+            String xFapiInteractionId,
+            String xCustomerUserAgent,
+            String apiClientId,
+            HttpServletRequest request,
+            Principal principal) throws OBErrorResponseException {
+
+        logger.info("Processing getDomesticScheduledPaymentConsentsConsentId request - consentId: {}, apiClient: {}, x-fapi-interaction-id: {}",
```

Comment:
trace-level :-)

Reply:
Only need an overview rather than granular level info.

Reply:
For clarity, customers often set their logging to INFO.This means they'll _always_ see this log message. Is that what you expect?

However, to me, this log message looks like it is granluar - i.e. you only want it if there's a problem (where admin would set the system log level to TRACE or DEBUG to follow what's going on under the hood). I would expect it's only in that scenario that we'd want to log this - so TRACE would be better.

Reply:
Changed log level. :)